components/resources/NewResourceForm.js

```diff
@@ -1,53 +1,59 @@
-import React, { useState, useCallback } from "react";
+import React from "react";
 import { TextField, Button, Stack } from "@mui/material";
+import { useFormik } from "formik";
+import * as yup from "yup";
 
 function NewResourceForm({ handleClose, onSubmit }) {
-  const [formInput, setContent] = useState({ name: "", url: "http://" });
+  const validationSchema = yup.object({
+    name: yup
+      .string("Enter a name")
+      .required("You can't submit a resource without a name"),
```

Comment:
If possible, add a restriction that name should have letters, because now it is possible to submit a name that is only blanks or only digits.